### PR TITLE
Fix multiprocessing hang on FSErrors

### PR DESCRIPTION
**Pull request by Spacerat**
This is a fix for the problem of subclasses of FSError/Exception causing multiprocessing to hang ( issue #183 ) if they change the arguments to `__init__`.  I went with @willmcgugan 's suggestion to implement `__reduce__` in each subclass wherever it was required.

### CODE REVIEW

File: fs/errors.py [hunk 1]
```diff
@@ -59,11 +59,15 @@ class MissingInfoNamespace(AttributeError):
 
     def __init__(self, namespace):
         # type: (Text) -> None
+        self.namespace=namespace
```
Comment (willmcgugan) [inline fs/errors.py]: Tiny pep8 thing, should be whitespace around `=`

**Comment by willmcgugan** [comment]
@Spacerat Looks good, thanks. Just a tiny request in the comments, and its good to merge.

**Comment by Spacerat** [comment]
:+1: thanks for the review @willmcgugan . I've fixed the formatting error now

**Comment by willmcgugan** [comment]
Thanks.

**Pull request merged**